libp2p/protocols/relay/client.nim

```diff
@@ -0,0 +1,266 @@
+## Nim-LibP2P
+## Copyright (c) 2022 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+import chronos, chronicles, options
+import ./messages,
+       ./rconn,
+       ./utils,
+       ../../peerinfo,
+       ../../switch,
+       ../../multiaddress,
+       ../../stream/connection
+
+import std/times
+
+logScope:
+  topics = "libp2p circuit relay client"
+
+const ClientMsgSize = 4096
+
+type
+  ClientError* = object of LPError
+  ReservationError* = object of ClientError
+  RelayV1DialError* = object of ClientError
+  RelayV2DialError* = object of ClientError
+  ClientAddConn* = proc(conn: Connection,
+                        duration: uint32,
+                        data: uint64): Future[void] {.gcsafe, raises: [Defect].}
+  Client* = ref object of LPProtocol
```

Comment:
`Client` is a too generic name for something exposed to users, consider renaming to `RelayClient`

Also, mounting the RelayV1 twice (once here and once in `Relay`) is going to cause issues when you want to both be a client & server